src/components/cc-domain-management/cc-domain-management.smart.js

```diff
@@ -0,0 +1,396 @@
+import '../cc-smart-container/cc-smart-container.js';
+
+/**
+ * @typedef {import('./cc-domain-management.types.js').DomainManagementFormState} DomainManagementFormState
+ * @typedef {import('./cc-domain-management.types.js').DomainManagementListStateLoaded} DomainManagementListStateLoaded
+ * @typedef {import('./cc-domain-management.types.js').DomainStateIdle} DomainStateIdle
+ * @typedef {import('./cc-domain-management.types.js').DomainState} DomainState
+ * @typedef {import('./cc-domain-management.types.js').DomainInfo} DomainInfo
+ * @typedef {import('../../lib/send-to-api.js').ApiConfig} ApiConfig
+ * @typedef {{ fqdn: string }} RawDomainFromApi
+ * @typedef {Omit<DomainInfo, 'isPrimary'>} DomainInfoWithoutIsPrimary
+ */
+
+defineSmartComponent({
+  selector: 'cc-domain-management',
+  params: {
+    apiConfig: { type: Object },
+    appId: { type: String },
+    ownerId: { type: String },
+  },
+  onContextUpdate ({ context, onEvent, updateComponent, signal }) {
+    const { apiConfig, appId, ownerId } = context;
+
+    /**
+     * @param {string} id
+     * @param {(domainState: DomainState) => void} callback
+     **/
+    function updateDomain (id, callback) {
+      updateComponent('domainListState',
+        /** @param {DomainManagementListStateLoaded} domainListState */
+        (domainListState) => {
+          const domainState = domainListState.domains.find((domain) => domain.id === id);
+          if (domainState != null) {
+            callback(domainState);
+          }
+        });
+    }
+
+    getDomains({ apiConfig, ownerId, appId, signal })
+      .then((domains) => {
+        updateComponent('domainListState', { type: 'loaded', domains });
+      })
+      .catch((error) => {
+        console.error(error);
+        updateComponent('domainListState', { type: 'error' });
+      });
+
+    fetchDnsInfo({ apiConfig, ownerId, appId, signal })
+      .then(({ cnameRecord, aRecords }) => {
+        updateComponent('dnsInfoState', { type: 'loaded', cnameRecord, aRecords });
+      })
+      .catch((error) => {
+        console.error(error);
+        updateComponent('dnsInfoState', { type: 'error' });
+      });
+
+    onEvent('cc-domain-management:add',
+      /** @param {{ hostname: string, pathPrefix: string, isWildcard: boolean }} domain */
+      ({ hostname, pathPrefix, isWildcard }) => {
+        const domainWithPathAndWildcard = getHostWithWildcard(hostname + pathPrefix, isWildcard);
+        updateComponent('domainFormState',
+          /** @param {DomainManagementFormState} domainFormState */
+          (domainFormState) => {
+            domainFormState.type = 'adding';
+          });
+
+        createNewDomain({ apiConfig, ownerId, appId, hostname, pathPrefix, isWildcard })
+          .then(() => {
+            if (isTestDomain(hostname)) {
+              notifySuccess(i18n('cc-domain-management.form.submit.success', { domain: domainWithPathAndWildcard }));
```

Comment:
question: the message is the same if it's a test domain or not. The only change seems to be the options for the notice and a `p` for the other translated message/notice below. Could you explain why the behavior is not the same? I'm not really sure I understand the point. :thinking:

Reply:
It should behave as follows: 
- if the domain is a custom domain, the info message tells you to config and we want the message to remain visible because the domain association won't work unless you configure your DNS (same idea as when we leave the message visible when  you add a secondary email because it needs to be validated).
- if the domain is a test domain, there's nothing to do, it just works so we go for a simple success message.